EngineMath/Vector.cpp

```diff
@@ -8,4 +8,76 @@
 
 namespace EngineMath {
 
+	//Public CTORS
+	Vector::Vector() {
+		data = new T[size];
+		for (int i = 0; i < size; i++)
+			data[i] = 0;
+	}
+
+	Vector::Vector(Ts... args) {
+		data = new T[size];
+		data = {args...};
+	}
+
+	//Public Methods
+
+	Vector::Dot(const Vector& other) {
+		if (size != other.size)
+			//ERROR
+		else {
+			auto sum = data[0] * other.data[0];
+			for (int i = 1; i < size; i++)
+				sum += data[i] * other.data[i];
+
+			return sum;
+		}
+	}
+
+	Vector::Project(const Vector & other) {
+			return other * (this.Dot(other) / other.Dot(other));
```

Comment:
I love the syntax here. Very clean.

Won't compile though.

this.Dot(other) needs to read this->Dot(other) because "this" is a pointer